### PR TITLE
Enable USE_UMFPACK in surface module on macOS CI

### PR DESCRIPTION
**Pull request by cybaol**

### CODE REVIEW

File: CMakeLists.txt
Comment (larshg) [inline CMakeLists.txt:296]: Can you remove this change from this PR?

File: surface/src/3rdparty/opennurbs/opennurbs_rtree.cpp
Comment (larshg) [inline surface/src/3rdparty/opennurbs/opennurbs_rtree.cpp:1475]: Lets just remove this, if you think its not of use?

**Comment by larshg** [comment]
Since it only touches 3. party On nurbs, I think its fine to merge (our CI's is green) - what do you thing @mvieth ?
But I won't be able to review in detail.

@cybaol Do you have any benchmarks for how much its improved?

File: surface/src/3rdparty/opennurbs/opennurbs_rtree.cpp
Comment (cybaol) [reply surface/src/3rdparty/opennurbs/opennurbs_rtree.cpp:1475]: Of course

File: CMakeLists.txt
Comment (cybaol) [reply CMakeLists.txt:296]: If this is removed, SuiteSparse can't find OpenMP, and I've moved this to #6114 , but there are still some issues and details to be discussed. Merge this after the #6114 merge?

Comment (larshg) [reply CMakeLists.txt:296]: What if you try remove the local findOpenMP and use the one from Cmake itself, as in #6100?

**Comment by cybaol** [comment]
> @cybaol Do you have any benchmarks for how much its improved?

No, I just found USE_UMFPACK is false in surface/on_nurbs(see https://github.com/PointCloudLibrary/pcl/blob/master/surface/src/on_nurbs/on_nurbs.cmake#L50). And didn't bother to build it in any of the other CIs.
So I decided to test it on macOS CI.

But, I've found that clang's stricter code checking has led to a lot of compilation errors in third-party opennurs library. So I fixed it.

File: surface/src/3rdparty/opennurbs/opennurbs_object.cpp
Comment (mvieth) [inline surface/src/3rdparty/opennurbs/opennurbs_object.cpp:843]: Can you replace with these lines, please: https://github.com/mcneel/opennurbs/blob/8.x/opennurbs_object.cpp#L806-L808 As far as I understand, these make sure that the string `p->m_sBaseClassName` ends with a null character/terminating character, but is not an empty string.

File: surface/src/3rdparty/opennurbs/opennurbs_nurbssurface.cpp
Comment (mvieth) [inline surface/src/3rdparty/opennurbs/opennurbs_nurbssurface.cpp:146]:
```suggestion
  if ( m_order[dir] >= 2 && m_cv_count[dir] >= m_order[dir] && nullptr != m_knot[dir] && t0 < t1 ) {
```
See https://github.com/mcneel/opennurbs/blob/8.x/opennurbs_nurbssurface.cpp#L149

File: surface/src/3rdparty/opennurbs/opennurbs_mesh.cpp
Comment (mvieth) [inline surface/src/3rdparty/opennurbs/opennurbs_mesh.cpp:3692]: Let's remove merge_count completely instead of just commenting out

File: surface/src/3rdparty/opennurbs/opennurbs_3dm_settings.cpp
Comment (mvieth) [inline surface/src/3rdparty/opennurbs/opennurbs_3dm_settings.cpp:3052]: Please remove chunk_count completely

**Review by mvieth — APPROVED** [review]
Looks good, thanks. Since Lars already gave his okay, I will merge immediately.

**Pull request merged**
